rules/Privatization/NodeManipulator/VisibilityManipulator.php

```diff
@@ -157,7 +157,7 @@ private function replaceVisibilityFlag(ClassMethod | Property | ClassConst $node
         $this->addVisibilityFlag($node, $visibility);
 
         if ($isStatic) {
-            $this->makeStatic($node);
+            $this->addVisibilityFlag($node, Visibility::STATIC);
```

Comment:
Why this change? Now we have to maintain the same code on 2 places, which might lead to change of one without another.

Reply:
I just tried to be as close to the previous code as possible without breaking other functionality.
We need to apparently strip the static tag so the visibility is correctly adjusted. (If I would not strip it other tests are failing)
What do you exactly mean by "maintain the same coe on 2 places"? Maybe I can find a way to improve this.

Reply:
The code seems 100 % same as in `makeStatic()`.

> What do you exactly mean by "maintain the same coe on 2 places"? Maybe I can find a way to improve this.

If there is bug in `makeStatic()`, now we have to fix in 2 places.
How does this fix the reported bug?

Reply:
I honestly copied from makeStatic and forgot that I did that when trying to look for the solution 😅 Adjusted the calls to use the makeStatic and makeNonStatic functions instead.

Previously we were removing the visibility modifier (which I don't quite understand why) because it somehow fixed https://github.com/rectorphp/rector/issues/4941 in https://github.com/rectorphp/rector/pull/4951 but it looks like this change is actually the only one required to fix the problem listed in rectorphp/rector#6829.
I didn't dig deep enough to find out why this is the case though.